File: 0/Templates/CSharp/ProjectTemplates/SplitViewAppTemplate/Presentation/SettingsViewModel.cs
using Intense.Presentation;
using Intense.UI;
using System;
using System.Collections.Generic;
using System.Collections.Immutable;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Windows.UI.Xaml;
using Windows.UI.Xaml.Media;

namespace $safeprojectname$.Presentation
{
    /// <summary>
    /// A sample settings view model.
    /// </summary>
    public class SettingsViewModel
        : NotifyPropertyChanged
    {
        private DisplayableTheme selectedTheme;
        private SolidColorBrush selectedBrush;
        private bool useSystemAccentColor;

        /// <summary>
        /// Initializes a new instance of the <see cref="SettingsViewModel"/>.
        /// </summary>
        public SettingsViewModel()
        {
            this.Brushes = AccentColors.Windows10.Select(c => new SolidColorBrush(c)).ToImmutableList();
            this.Themes = ImmutableList.Create(
                new DisplayableTheme("Dark", ApplicationTheme.Dark),
                new DisplayableTheme("Light", ApplicationTheme.Light));

            // ensure viewmodel state reflects actual appearance
            var manager = AppearanceManager.GetForCurrentView();
            this.selectedTheme = this.Themes.FirstOrDefault(t => t.Theme == manager.Theme);

            if (AppearanceManager.AccentColor == null)
            {
                this.useSystemAccentColor = true;
            }
            else
            {
                this.selectedBrush = this.Brushes.FirstOrDefault(b => b.Color == AppearanceManager.AccentColor);
            }
        }

        /// <summary>
        /// Gets the brushes.
        /// </summary>
        public IReadOnlyList<SolidColorBrush> Brushes { get; }

        /// <summary>
        /// Gets or sets the selected brush.
        /// </summary>
        public SolidColorBrush SelectedBrush
        {
            get { return this.selectedBrush; }
            set
            {
                if (Set(ref this.selectedBrush, value) && !this.useSystemAccentColor && value != null)
                {
                    AppearanceManager.AccentColor = value.Color;
                }
            }
        }

        /// <summary>
        /// Gets the collection of themes.
        /// </summary>
        public IReadOnlyList<DisplayableTheme> Themes { get; }
        /// <summary>
        /// Gets or sets the selected theme.
        /// </summary>
        public DisplayableTheme SelectedTheme
        {
            get { return this.selectedTheme; }
            set
            {
                if (Set(ref this.selectedTheme, value) && value != null)
                {
                    AppearanceManager.GetForCurrentView().Theme = value.Theme;
                }
            }
        }

        /// <summary>
        /// Gets or sets a value indicating whether to use the system accent color.
        /// </summary>
        public bool UseSystemAccentColor
        {
            get { return this.useSystemAccentColor; }
            set
            {
                if (Set(ref this.useSystemAccentColor, value))
                {
                    if (value)
                    {
                        AppearanceManager.AccentColor = null;
                    }
                    else if (this.SelectedBrush != null) 
                    {
                        AppearanceManager.AccentColor = this.SelectedBrush.Color;
                    }
                }
            }
        }
    }
}

File: 0/Templates/VisualBasic/ProjectTemplates/SplitViewAppVBTemplate/Presentation/SettingsViewModel.vb
Imports System.Collections.Immutable
Imports Intense.Presentation
Imports Intense.UI

Namespace Presentation
    Public Class SettingsViewModel
        Inherits NotifyPropertyChanged

        Private _themes As IReadOnlyList(Of DisplayableTheme)
        Private _selectedTheme As DisplayableTheme
        Private _brushes As IReadOnlyList(Of SolidColorBrush)
        Private _selectedBrush As SolidColorBrush
        Private _useSystemAccentColor As Boolean

        Public Sub New()
            _brushes = AccentColors.Windows10.Select(Function(c)
                                                         Return New SolidColorBrush(c)
                                                     End Function).ToImmutableList()
            _themes = ImmutableList.Create(
                New DisplayableTheme("Dark", ApplicationTheme.Dark),
                New DisplayableTheme("Light", ApplicationTheme.Light))

            Dim manager = AppearanceManager.GetForCurrentView()
            _selectedTheme = _themes.FirstOrDefault(Function(t)
                                                        Return t.Theme = manager.Theme
                                                    End Function)

            If (AppearanceManager.AccentColor Is Nothing) Then
                _useSystemAccentColor = True
            Else
                _selectedBrush = _brushes.FirstOrDefault(Function(b)
                                                             Return b.Color = AppearanceManager.AccentColor
                                                         End Function)
            End If
        End Sub

        Public ReadOnly Property Brushes As IReadOnlyList(Of SolidColorBrush)
            Get
                Return _brushes
            End Get
        End Property

        Public Property SelectedBrush As SolidColorBrush
            Get
                Return _selectedBrush
            End Get
            Set
                If Me.Set(_selectedBrush, Value) And Not _useSystemAccentColor And Not Value Is Nothing Then
                    AppearanceManager.AccentColor = Value.Color
                End If
            End Set
        End Property

        Public ReadOnly Property Themes As IReadOnlyList(Of DisplayableTheme)
            Get
                Return _themes
            End Get
        End Property

        Public Property SelectedTheme As DisplayableTheme
            Get
                Return _selectedTheme
            End Get
            Set
                If Me.Set(_selectedTheme, Value) And Not Value Is Nothing Then
                    AppearanceManager.GetForCurrentView().Theme = Value.Theme
                End If
            End Set
        End Property

        Public Property UseSystemAccentColor As Boolean
            Get
                Return _useSystemAccentColor
            End Get
            Set
                If Me.Set(_useSystemAccentColor, Value) Then
                    If (Value) Then
                        AppearanceManager.AccentColor = Nothing
                    ElseIf Not _selectedBrush Is Nothing Then
                        AppearanceManager.AccentColor = _selectedBrush.Color
                    End If
                End If
            End Set
        End Property

    End Class
End Namespace



